Sources/PackageDescription/Version.swift

```diff
@@ -25,13 +25,17 @@ public struct Version {
     }
 
     public init?(_ characters: String.CharacterView) {
-        let components = characters.split(".", maxSplit: 2, allowEmptySlices: true).map(String.init).flatMap{ Int($0) }.filter{ $0 >= 0 }
+        let components = Version.componentsFromCharacters(characters)
         guard components.count == 3 else { return nil }
 
         self.major = components[0]
         self.minor = components[1]
         self.patch = components[2]
     }
+    
+    private static func componentsFromCharacters(characters: String.CharacterView) -> [Int] {
+        return characters.split(".", maxSplit: 2, allowEmptySlices: true).flatMap{ Int(String($0)) }.filter{ $0 >= 0 }
```

Comment:
Actually, the string ".1." will return `[1]`, which the `Version` `init` handles because it requires there to be three elements, but the `Specifier` `init` doesn't handle correctly. The `Specifier` `init` will assign `1` to `major` and `nil` to the others in that case.